n3fit/src/n3fit/hyper_optimization/mongofiletrials.py

```diff
@@ -0,0 +1,290 @@
+        self.db_port = str(db_port)
+        self.db_name = db_name
+        self.num_workers = num_workers
+        self.mongotrials_arg = f"mongo://{self.db_host}:{self.db_port}/{self.db_name}/jobs"
+        self.workers = []
+
+        self._store_trial = False
+        self._json_file = replica_path / "tries.json"
+        self.database_tar_file = replica_path / f"{self.db_name}.tar.gz"
+        self._parameters = parameters
+        self._rstate = None
+        self._dynamic_trials = []
+
+        super().__init__(self.mongotrials_arg, *args, **kwargs)
+
+    @property
+    def rstate(self):
+        """Returns the rstate attribute; see :class:`n3fit.hyper_optimization.filetrials.FileTrials`."""
+        return self._rstate
+
+    @rstate.setter
+    def rstate(self, random_generator):
+        """Sets the rstate attribute; see :class:`n3fit.hyper_optimization.filetrials.FileTrials`."""
+        self._rstate = random_generator
+
+    def _set_dynamic_trials(self):
+        """Converts self._trials to a dictionary and stores it in self._dynamic_trials."""
+        self._dynamic_trials = [convert_bson_to_dict(item) for item in self._trials]
+
+    def refresh(self):
+        """Fetches data from mongo database and save to a json file."""
+        super().refresh()
+
+        # convert BSON object to a dictionary
+        self._set_dynamic_trials()
+
+        # write json to disk
+        if self._store_trial:
+            log.info("Storing scan in %s", self._json_file)
+            local_trials = []
+            for idx, t in enumerate(self._dynamic_trials):
+                local_trials.append(t)
+                local_trials[idx]["misc"]["space_vals"] = space_eval_trial(self._parameters, t)
+
+            all_to_str = json.dumps(local_trials, default=str)
+            with open(self._json_file, "w") as f:
+                f.write(all_to_str)
+
+    # like in `FileTrials` the two methods below are implemented to avoid writing to the database twice
+    def new_trial_ids(self, n):
+        self._store_trial = False
+        return super().new_trial_ids(n)
+
+    def _insert_trial_docs(self, docs):
+        self._store_trial = True
+        return super()._insert_trial_docs(docs)
+
+    def start_mongo_workers(
+        self, workdir=None, exp_key=None, poll_interval=0.1, use_subprocesses=False
+    ):
+        """Initiates all mongo workers simultaneously."""
+        # get the number of gpu cards, if any
+        gpus_all_physical_list = get_physical_gpus()
+        num_gpus_available = len(gpus_all_physical_list)
+        if not num_gpus_available:
+            log.warning("No GPUs found in the system.")
+
+        # launch mongo workers
+        for i in range(self.num_workers):
+            # construct the command to start a hyperopt-mongo-worker
+            args = [
+                "hyperopt-mongo-worker",
+                "--mongo",
+                f"{self.db_host}:{self.db_port}/{self.db_name}",
+            ]
```

Comment:
```suggestion
                f"{self.db_host}:{self.db_port}/{self.db_name}",
                 "--max-consecutive-failures", "10",
                 "--reserve-timeout", "600",
            ]
```

This fixes the timeout issue we had, not sure if one of the two and/or lower settings would suffice, but this was enough for the hyperopt runcard, and I think there's no harm in having these higher than necessary.

Reply:
Thanks @APJansen. I have added these options to the [start_mongo_workers](https://github.com/NNPDF/nnpdf/blob/5628cba72bcac6d3ac226b2b1d2b8bce821cb748/n3fit/src/n3fit/hyper_optimization/mongofiletrials.py#L217) method but opted for now to leave their values on the defaults.

Reply:
I made some tests and it appears to have no downsides. I added these values as default.